addons/refuel/XEH_postInit.sqf

```diff
@@ -1,9 +1,84 @@
 #include "script_component.hpp"
+if (!GVAR(enabled)) exitWith {};
+
+["All", "InitPost", {
+    params ["_vehicle"];
+    if (getFuelCargo _vehicle <= 0) exitWith {};
+    TRACE_1("initPost",_vehicle);
+
+    if (local _vehicle) then {
+        _vehicle setFuelCargo 0;
+        LOG("initPost setFuelCargo");
+    };
+    if (-1 == _vehicle getVariable [QGVAR(HDEHID), -1]) then {
+        _vehicle setVariable [QGVAR(HDEHID), _vehicle addEventHandler ["HandleDamage", LINKFUNC(handleDamage)]];
+        LOG("initPost add HDEH");
+    };
+}, true, ["Man"], true] call CBA_fnc_addClassEventHandler;
 
 if (isServer) then {
     addMissionEventHandler ["HandleDisconnect", {call FUNC(handleDisconnect)}];
 };
 
+private _cfgPositions = configFile >> QGVAR(positions) >> worldName;
+if (isArray _cfgPositions) then {
+    {
+        _x params ["_class", "_positions"];
+        {
+            private _objects = _x nearObjects [_class, 30];
+            if (_objects isEqualTo []) then {
+                WARNING_3("no pumps %1 found near %2 %3",_class,worldName,_x);
+            } else {
+                {
+                    // terrain fuel pumps don't trigger init and must setFuelCargo on each client
+                    _x setFuelCargo 0;
+                    if (isServer) then {
+                        _x addEventHandler ["HandleDamage", LINKFUNC(handleDamage)];
+                        _x setVariable [QGVAR(isTerrainPump), true];
+                    };
+                } forEach _objects;
+            };
+        } forEach _positions;
+    } forEach getArray _cfgPositions;
+
+    // placed in editor static objects don't trigger init but synchronize fuel cargo
+    // placed in editor vehicles both trigger init and synchronize fuel cargo
+    {
+        if (getFuelCargo _x > 0) then {
+            TRACE_1("allMissionObjects",_x);
+            if (local _x) then {
+                _x setFuelCargo 0;
+                LOG("allMissionObjects setFuelCargo");
+            };
+            _x setVariable [QGVAR(HDEHID), _x addEventHandler ["HandleDamage", LINKFUNC(handleDamage)]];
+        };
+    } forEach allMissionObjects "";
+} else {
+    // here are both terrain and editor static objects
+    WARNING_2("World %1: %2 is not configured; can load slower",worldName,QGVAR(positions));
+    private _halfWorldSize = worldSize / 2;
+    private _worldCenter = [_halfWorldSize, _halfWorldSize];
+    _halfWorldSize = _halfWorldSize * sqrt 2;
+    private _refuelMissionObjects = allMissionObjects "" select {getFuelCargo _x > 0};
```

Comment:
use https://community.bistudio.com/wiki/allObjects instead?

Reply:
On Altis `-1 allObjects 0` is about 30% slowed than `allMissionObjects ""`.

Reply:
`9 allObjects 0`?

Reply:
`9 allObjects 0` is about 12% faster than `allMissionObjects ""`, although `allMissionObjects "all"` is even faster than `9 allObjects 0`, albeit by a small margin.

I'm a bit worried about the note on the `allObjects` wiki page:

> Some terrain placed objects are not part of slow vehicles collection, use [nearestTerrainObjects](https://community.bistudio.com/wiki/nearestTerrainObjects) command to return these objects

That means there are some buildings that are part of it - which ones? We don't want terrain buildings, but do we want buildings that were placed down in the editor?

Reply:
`allMissionObjects` returns only editor placed objects. Usually on init there are not so many objects. Why do you want to replace it with `allObjects`?